integration/nwo/deploy.go

```diff
@@ -220,13 +220,12 @@ func InstallChaincodeLegacy(n *Network, chaincode Chaincode, peers ...*Peer) {
 }
 
 func ApproveChaincodeForMyOrg(n *Network, channel string, orderer *Orderer, chaincode Chaincode, peers ...*Peer) {
-	if chaincode.PackageID == "" {
-		chaincode.SetPackageIDFromPackageFile()
-	}
-
 	// used to ensure we only approve once per org
 	approvedOrgs := map[string]bool{}
 	for _, p := range peers {
+		if chaincode.PackageID == "" && !p.DevMode {
```

Comment:
Taking the bigger picture into account, I think we shouldn't have this check for DevMode in this nwo helper function for approving the chaincode definition. I'd only expect devmode to be utilized in the devmode suite so I think it makes the most sense to simply include a similar helper function there and keep the nwo package's helper function streamlined.